File: Server/src/main/java/com/example/server/grpc/converters/TeamConverter.java
package com.example.server.grpc.converters;

import com.example.server.dto.TeamDTO;
import grpcProtoFiles.TeamOuterClass;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public class TeamConverter {

    private PlayerConverter playerConverter;

    @Autowired
    public TeamConverter(PlayerConverter playerConverter) {
        this.playerConverter = playerConverter;
    }

    public TeamDTO toDto(TeamOuterClass.Team team) {
        return TeamDTO.builder()
                .name(team.getName())
                .players(playerConverter.toDtoList(team.getPlayerList()))
                .build();
    }

    public List<TeamDTO> toDtoList(List<TeamOuterClass.Team> teams) {
        return teams.stream().map(team -> toDto(team)).toList();
    }

    public TeamOuterClass.Team toProto(TeamDTO team) {
        TeamOuterClass.Team.Builder builder = TeamOuterClass.Team.newBuilder()
                .setName(team.getName());

        team.getPlayers().forEach(player -> builder.addPlayer(playerConverter.toProto(player)));

        return builder.build();
    }

    public List<TeamOuterClass.Team> toProtoList(List<TeamDTO> teams) {
        return teams.stream().map(team -> toProto(team)).toList();
    }
}

File: DataServer/GrpcService/converters/TeamConverter.cs
using DTO;
using Google.Protobuf.Collections;
using grpcProtoFiles;

namespace GrpcService.converters;

public class TeamConverter
{
    
    private readonly PlayerConverter playerConverter;

    public TeamConverter(PlayerConverter playerConverter)
    {
        this.playerConverter = playerConverter;
    }
    
    public TeamDTO ToDTO(Team team) {
        return new TeamDTO {
            Name = team.Name,
            Players = playerConverter.ToDTOList(team.Players.ToList())
        };
    }

    public List<TeamDTO> ToDTOList(List<Team> teams)
    {
        List<TeamDTO> _teams = new List<TeamDTO>();
        foreach (var team in teams)
        {
            _teams.Add(ToDTO(team));
        }

        return _teams;
    }

    public Team ToProto(TeamDTO team) {
        return new Team {
            Name = team.Name,
            Players = { playerConverter.ToProtoList(team.Players) }
        };
    }
    
    public RepeatedField<Team> ToProtoList(List<TeamDTO> teams)
    {
        List<Team> _teams = new List<Team>();
        foreach (var team in teams)
        {
            _teams.Add(ToProto(team));
        }

        return new RepeatedField<Team>(){_teams};
    }
}